### ISSUE TITLE
Rails 7: "uninitialized constant ActionText::RichTextDashboard"

### ISSUE THREAD

**Issue by arenzel** [thoughtbot/administrate#2175]
## What were you trying to do?

I'm trying to create a new `Post` from a standard installation of Administrate on Rails 7.
When hitting `http://localhost:3000/admin/posts/new`, I'm getting the following error:

```
NameError in Admin::Posts#new

Showing /<path>/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/administrate-0.17.0/app/views/fields/has_one/_form.html.erb where line #19 raised:

uninitialized constant ActionText::RichTextDashboard

      Object.const_get(camel_cased_word)
            ^^^^^^^^^^
Did you mean?  RichTextAreaField

[...]

Trace of template inclusion: #<ActionView::Template /Users/arnaudlenglet/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/administrate-0.17.0/app/views/administrate/application/_form.html.erb locals=["page"]>, #<ActionView::Template /Users/arnaudlenglet/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/administrate-0.17.0/app/views/administrate/application/new.html.erb locals=["page"]>
```

### The Post model
```rb
# == Schema Information
#
# Table name: posts
#
#  id           :integer          not null, primary key
#  title        :string
#  author       :string
#  except       :string
#  content      :text
#  published_at :datetime
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#
class Post < ApplicationRecord
  validates_presence_of :title, :author

  has_rich_text :content

  has_many :comments, dependent: :destroy
end
```

I can confirm this is somehow related to RichText because I don't have the issue when visiting other models that don't have any RichText attribute.

I've already tried this command (from this [wiki page]($ rails g administrate:field rich_text_area)), but didn't solved it:
```
$ rails g administrate:field rich_text_area
```

## What versions are you running?
  - Rails Rails 7.0.2.3
  - administrate 0.17.0

**Comment by MuhammetDilmac** [comment]
Use `RichTextAreaField` instead of `ActionText::RichTextDashboard` in attribute types.

Like this;
```ruby
ATTRIBUTE_TYPES = {
  id: Field::String,
  title: Field::String,
  author: Field::String,
  except: Field::String
  content: RichTextAreaField,
  published_at: Field::DateTime,
  created_at: Field::DateTime,
  updated_at: Field::DateTime
}
```

**Comment by arenzel** [comment]
Thank you for your reply @MuhammetDilmac.
I don't have any `ActionText::RichTextDashboard` in attribute_types.

Example for the `Collection` model:

```rb
# File: app/controllers/admin/dashboards/collection_dashboard

ATTRIBUTE_TYPES = {
    rich_text_description: Field::HasOne,

    [...]

    description: Field::Text,   # <---- Rich text field here
    name: Field::String,
    author: Field::String,
  }.freeze
```

When I try to replace `description: Field::Text` by `description: RichTextAreaField ` I get the following error:

```
uninitialized constant CollectionDashboard::RichTextAreaField
```

**Comment by arenzel** [comment]
# Issue solved! 🎉 

My bad: I was misusing the custom field. 🤦 
For my fellow developers who will encounter the same issue:

## Step by step to fix it

> 💡  Assuming you have a model `Post` with a RichText field named `description`.

1. Create a new custom **RichTextAreaField** by following the instruction from this [wiki page](https://github.com/thoughtbot/administrate/wiki/Field:-RichTextAreaField) (+ more info on custom fields [in the docs here](https://administrate-demo.herokuapp.com/adding_custom_field_types))
2.  In `post_dashboard.rb`, within the **ATTRIBUTE_TYPES** hash, replace the type of the `description` attribute by `RichTextAreaField`.
3.  Then, at the place, also remove the attribute `rich_text_description` (was generated by the generator but that's an error)

To summarize:

```rb
# app/dashboards/post_dashboard.rb

ATTRIBUTE_TYPES = {
    [...]
    rich_text_description: Field::HasOne,    # <--- DELETE THIS LINE
    description: Field::Text                 # <--- UPDATE THIS LINE
  }.freeze
```

Hope it helps! 🤗

**Comment by sebastienbourda** [comment]
> # Issue solved! 🎉
> My bad: I was misusing the custom field. 🤦 For my fellow developers who will encounter the same issue:
> 
> ## Step by step to fix it
> > 💡  Assuming you have a model `Post` with a RichText field named `description`.
> 
> 1. Create a new custom **RichTextAreaField** by following the instruction from this [wiki page](https://github.com/thoughtbot/administrate/wiki/Field:-RichTextAreaField) (+ more info on custom fields [in the docs here](https://administrate-demo.herokuapp.com/adding_custom_field_types))
> 2. In `post_dashboard.rb`, within the **ATTRIBUTE_TYPES** hash, replace the type of the `description` attribute by `RichTextAreaField`.
> 3. Then, at the place, also remove the attribute `rich_text_description` (was generated by the generator but that's an error)
> 
> To summarize:
> 
> ```ruby
> # app/dashboards/post_dashboard.rb
> 
> ATTRIBUTE_TYPES = {
>     [...]
>     rich_text_description: Field::HasOne,    # <--- DELETE THIS LINE
>     description: Field::Text                 # <--- UPDATE THIS LINE
>   }.freeze
> ```
> 
> Hope it helps! 🤗

Thanks for the fix!
